tests/client.cpp

```diff
@@ -508,76 +508,73 @@ TEST_F(ConnectTest, Success) {
   ASSERT_MEMORY_OK();
 }
 
-TEST_F(ConnectTest, Success_v4) {
-  RunServer([](int sockfd) {
-    // Perform handshake.
-    {
-      char handshake[20];
-      ASSERT_EQ(RecvData(sockfd, handshake, 20), 0);
-      ASSERT_EQ(std::string(handshake, 4), "\x60\x60\xB0\x17"s);
-      ASSERT_EQ(std::string(handshake + 4, 4), "\x00\x00\x01\x04"s);
-      ASSERT_EQ(std::string(handshake + 8, 4), "\x00\x00\x00\x01"s);
-      ASSERT_EQ(std::string(handshake + 12, 4), "\x00\x00\x00\x00"s);
-      ASSERT_EQ(std::string(handshake + 16, 4), "\x00\x00\x00\x00"s);
+auto run_v4_server_success = [](int sockfd) {
+  // Perform handshake.
+  {
+    char handshake[20];
+    ASSERT_EQ(RecvData(sockfd, handshake, 20), 0);
+    ASSERT_EQ(std::string(handshake, 4), "\x60\x60\xB0\x17"s);
+    ASSERT_EQ(std::string(handshake + 4, 4), "\x00\x00\x01\x04"s);
+    ASSERT_EQ(std::string(handshake + 8, 4), "\x00\x00\x00\x01"s);
+    ASSERT_EQ(std::string(handshake + 12, 4), "\x00\x00\x00\x00"s);
+    ASSERT_EQ(std::string(handshake + 16, 4), "\x00\x00\x00\x00"s);
 
-      uint32_t version = htobe32(0x0104);
-      ASSERT_EQ(SendData(sockfd, (char *)&version, 4), 0);
-    }
+    uint32_t version = htobe32(1);
```

Comment:
why changed from 0x0104 to 1?

Reply:
Fixed

Reply:
if `0x0104` is here the tests never finishes